2024/XenSummit/Challenges_and_Status_of_Enabling_TrenchBoot_in_Xen_Hypervisor.md

```diff
@@ -0,0 +1,322 @@
+3 phases.
+
+---
+
+# Phase 2 - TPM 2.0 support
+
+.center[.image-90[![](/img/tb_aem_phase2.png)]]
+
+???
+
+- Added TPM 2.0 support for dom0 kernel and initrd measurements
+- Added eventlog extraction and parsing
+- Implemented support for booting with Intel TXT and TPM 2.0 in legacy BIOS
+  boot mode
+
+---
+
+# Phase 3 - TrenchBoot boot protocol
+
+.center[.image-90[![](/img/tb_aem_phase3.png)]]
+
+???
+
+- Rebased changes onto the most recent TrenchBoot support patches to Linux
+  kernel introducing a unified TrenchBoot boot protocol
+- Final shape of patches is not yet determined, the work on finalizing the
+  boot protocol including UEFI boot mode is pending on the [Linux kernel
+  mailing list, currently patchset
+  v9](https://lkml.org/lkml/2024/5/30/1226)
+
+---
+
+# Phase 4 - AMD support
+
+.center[.image-80[![](/img/tb_aem_phase4.png)]]
+
+???
+
+- Implemented support for booting with AMD SKINIT and TPM 1.2/2.0 in legacy
+  BIOS boot mode
+- Implemented the newest TrenchBoot boot protocol in the Secure Kernel Loader
+- Simplified installation of the required packages by providing a RPM
+  repository for Qubes OS
+
+---
+
+# Try TrenchBoot
+
+It is relatively easy to get a hardware which supports DRTM:
+
+- Intel-based tested and known to work:
+  * Protectli VP4670 (open-source-firmware supported)
+  * Dell OptiPlex 7010/9010 (open-source-firmware supported)
+  * HP EliteDesk 800 G2 Desktop Mini
+
+.center[
+.image-20[![](/img/VP4600.png)]
+.image-10[![](/img/tb_aem_optiplex.png)]
+.image-20[![](/img/hp_elitedesk_800_g2.jpg)]
+]
+
+- AMD-based tested and known to work:
+  * Terminal HP T630
+
+.center[.image-20[![](/img/hp_t630.png)]]
+
+???
+
+These are the tested platforms that are working well. We have also faced
+issues with other platforms with BIOS bugs, which could not be workarounded,
+like Lenovo M920 or Supermicro M11SDV.
+
+Not all Intel platform support DRTM technology - Intel TXT. It requires a
+capable CPU and chipset. Typically vPro platforms satisfy these requirements,
+although it is still possible to have a Intel TXT initialization bug in BIOS,
+which prevents using DRTM unless a BIOS fix is issued.
+
+For AMD platforms, almost every CPU supports AMD SKINIT (CPU needs to support
+SVM basically and BIOS must not disable it).
+
+---
+
+# Current ongoing efforts
+
+- Provide package repositories for other major distros (Debian, Ubuntu Fedora)
+- Prepare tools for assessing the readiness of the system and BIOS to perform
+  Dynamic Launch
+- Adding DRTM Ready indicator to the system security reports: Qubes OS Security
+  Report, [fwupd HSI](https://fwupd.github.io/libfwupdplugin/hsi.html)
+
+.center[.image-40[![](/img/qubes-sec-report.png)]]
+
+---
+
+# Phase 5 - Xen as DLME for UEFI boot mode
+
+.center[.image-85[![](/img/tb_roadmap.png)]]
+
+More details about Secure Launch on [trenchboot.org](https://trenchboot.org/specifications/Secure_Launch/)
```

Comment:
This note seems to not be related to the rest of the slide.

Reply:
Yup, I removed it later